tests/core/pyspec/eth2spec/test/capella/block_processing/test_process_withdrawals.py

```diff
@@ -67,35 +95,81 @@ def test_success_empty_queue(spec, state):
 
 @with_capella_and_later
 @spec_state_test
-def test_success_one_in_queue(spec, state):
-    prepare_withdrawal_queue(spec, state, 1)
+def test_success_one_full_withdrawal(spec, state):
+    fully_withdrawable_indices, partial_withdrawals_indices = prepare_expected_withdrawals(
+        spec, state, num_full_withdrawals=1)
+    assert len(fully_withdrawable_indices) == 1
+    assert len(partial_withdrawals_indices) == 0
 
     next_slot(spec, state)
     execution_payload = build_empty_execution_payload(spec, state)
 
-    yield from run_withdrawals_processing(spec, state, execution_payload)
+    expected_withdrawals = yield from run_withdrawals_processing(spec, state, execution_payload)
```

Comment:
`run_*` should just encapsulate `verify_post_state` rather than having a return value in addition to it's yield. The default should be to do post-state validations rather than having to hoist it into another finction